### PR TITLE
Make businessKeys a process-wide concept

### PR DESCRIPTION
**Pull request by bravesirrobin81**

### CODE REVIEW

File: src/form/BusinessKeyVisitor.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export default class BusinessKeyVisitor {
+    visit(formComponent) {
+        const component = formComponent.component;
+        const dataResolveContext = formComponent.dataContext;
+
+        if (component.key === 'businessKey' && component.defaultValue === '') {
+            component.defaultValue = dataResolveContext.processContext.businessKey;
+        }
+    }
+}
```
Comment (shepda) [inline src/form/BusinessKeyVisitor.js]: Can we add some unit tests to cover this method?

File: src/services/DataContextFactory.js [hunk 2]
```diff
@@ -43,16 +44,24 @@ export default class DataContextFactory {
...
         }
     }
 
+    async createProcessContext(processData) {
+        const processContext = new ProcessContext(processData);
+        if (!processContext.businessKey) {
+          processContext.businessKey = await this.referenceGenerator.newBusinessKey();
+        }
+        return processContext;
+    }
```
Comment (shepda) [inline src/services/DataContextFactory.js]: Can we add some unit tests to cover this method?

File: src/form/BusinessKeyVisitor.js (hunk 1 above)
Comment (bravesirrobin81) [reply src/form/BusinessKeyVisitor.js]: These are covered by unit tests. see `formDataBusinessKey.test.js`

File: src/services/DataContextFactory.js (hunk 2 above)
Comment (bravesirrobin81) [reply src/services/DataContextFactory.js]: These are covered by unit tests. see `formDataBusinessKey.test.js`

**Pull request merged**